modules/default/weather/providers/envcanada.js

```diff
@@ -508,17 +501,17 @@ WeatherProvider.register("envcanada", {
 
 	setPrecipitation(weather, foreGroup, today) {
 		if (foreGroup[today].querySelector("precipitation accumulation")) {
-			weather.precipitation = foreGroup[today].querySelector("precipitation accumulation amount").textContent * 1.0;
+			weather.precipitationAmount = foreGroup[today].querySelector("precipitation accumulation amount").textContent * 1.0;
 
 			weather.precipitationUnits = " " + foreGroup[today].querySelector("precipitation accumulation amount").getAttribute("units");
 
 			if (this.config.units === "imperial") {
 				if (weather.precipitationUnits === " cm") {
-					weather.precipitation = (weather.precipitation * 0.394).toFixed(2);
+					weather.precipitationAmount = (weather.precipitationAmount * 0.394).toFixed(2);
```

Comment:
as a rule of thumb we dont want the weather providers to check for config.units and do conversions stuff based on that value.

instead all providers should store the data in metric units in the object (and the conversion based on the config happens only on the "frontend part" (aka in the njk file and when it gets the values from the weatherobject).

could you change the logic here and in other providers too?

Reply:
Great idea 👍 We now support mm (default) and cm as input units. This fixes the conversion for all providers that haven't bothered with the calculation in the first place.